sympy/solvers/ode.py

```diff
@@ -1044,6 +1037,47 @@ class in it.  Note that a hint may do this anyway if
             else:
                 raise ValueError("Enter boundary conditions of the form ics={f(point}: value, f(x).diff(x, order).subs(x, point): value}")
 
+
+    # Precondition to try remove unnecessary factor and power from equation
+    eqs = eq
+    if isinstance(eqs, Pow):
+        eqs_args = eqs.args # if f(x)**p=0 then f(x)=0 (p>0)
+        if eqs_args[1]>0:
+            eqs = eqs_args[0]
```

Comment:
Using `>` than can raise for symbolic conditions e.g. `f(x).diff(x)**n` where `n` is a symbol. What do you intend to do here if it is not possible to know whether the exponent is positive or not?

Reply:
If positive, then `Eq(eq**n, 0)` can be replaced by `Eq(eq, 0)` else Ieft as it is.

Now I am doing this because if ` eq=(f(x).diff(x)-f(x))**2` then We can solve easily by removing the power and this is a simple equation(maybe it is solvable after expanding) but for complex equation, we can't expand the whole equation. I will change the code if the power is not an integer.

Reply:
I understand but what I mean is that when you do `if x > 0:` then sometimes it will raise an exception e.g.:
```julia
In [30]: x > 0                                                                                                                    
Out[30]: x > 0

In [31]: if x > 0: pass                                                                                                           
---------------------------------------------------------------------------
TypeError: cannot determine truth value of Relational
```
If you don't want it to sometimes raise an exception you should do `if x.is_positive:`.

I also wonder if it would be better to check for powers earlier on in dsolve since this would also apply to non-factorable ODEs.

Reply:
Looks good to me. I will add in the next commit.